engine/src/main/java/io/camunda/zeebe/engine/processing/common/CatchEventBehavior.java

```diff
@@ -105,15 +105,24 @@ public void unsubscribeFromEvents(
   }
 
   private Either<Failure, EvalResult> evalExpressions(
-      final ExecutableCatchEvent event, final BpmnElementContext context) {
+      final ExpressionProcessor expressionProcessor,
+      final ExecutableCatchEvent event,
+      final BpmnElementContext context) {
     return Either.<Failure, EvalResult>right(new EvalResult(event))
-        .flatMap(result -> evaluateMessageName(event, context).map(result::messageName))
-        .flatMap(result -> evaluateCorrelationKey(event, context).map(result::correlationKey))
-        .flatMap(result -> evaluateTimer(event, context).map(result::timer));
+        .flatMap(
+            result ->
+                evaluateMessageName(expressionProcessor, event, context).map(result::messageName))
+        .flatMap(
+            result ->
+                evaluateCorrelationKey(expressionProcessor, event, context)
+                    .map(result::correlationKey))
+        .flatMap(result -> evaluateTimer(expressionProcessor, event, context).map(result::timer));
```

Comment:
🔧  I understand the motivation, but it does not improve the understandability of this code. We should consider how we can improve the readability here in addition to your improvement.

Reply:
Not sure what you are objecting to. It only adds a parameter to the call. In which way do you feel it alters readability?